mlibrary/lib.umich.edu
# Eslint9 fixes

# Overview
The repo was upgraded to the new major ESLint version 9. This PR fixes all the ESLint 9 issues under `/src` (~4,000). I believe every file has been changed.

Additional fixes and cleanup were done as part of this update as well. Four class components were converted to function components. This addresses [WEBSITE-96](https://mlit.atlassian.net/jira/software/projects/WEBSITE/list?filter=%22assignee%22%20%3D%20%275cd437964401890dcab382a4%27%20AND%20%22status%22%20%3D%20%27Done%27&selectedIssue=WEBSITE-96&text=component)
- src/reusable/alert.js
- src/reusable/expandable/expandable.js
- src/reusable/expandable/expandable-button.js
- src/reusable/expandable/expandable-children.js

> This pull request resolves/closes/fixes [WEBSITE-140](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-140).

## Anything else?
Rehype was once again reverted to the previous version. This update was fairly close to including the new Rehype / Unified changes but we were having an issue displaying images within paragraphs. I'm pulling that change out and bringing it to a separate ticket / branch.

The [Website updates testing list](https://mlit.atlassian.net/wiki/spaces/LDD/pages/9781346312/Website+updates+testing+list) provides a useful checklist of items to check and compare. 

## Testing
- Run `npx eslint ./src` and confirm there are no errors
- Make sure the PR is consistent in these browsers:
  - [] Safari
- Run accessibility tests:
